From fe801e6d77b69987c9cb066f5f9a5d597d06ca98 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Thu, 22 Mar 2018 15:53:24 -0500
Subject: [PATCH 3534/5242] MLK-17703-1: drm: imx: dcss: update HDR10 tables

commit  d0ff7135aaeb82abb9f45363b44d94effe1107b7 from
https://source.codeaurora.org/external/imx/linux-imx.git

The old tables had incorrect CSCBs when YUV formats were being used. That's
because the application used to generate the tables always assumed channel 0 is
graphics even if it was configured as YUV.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-hdr10-tables.h |  310 ++++++++++++++++++------------
 1 file changed, 182 insertions(+), 128 deletions(-)

diff --git a/drivers/gpu/imx/dcss/dcss-hdr10-tables.h b/drivers/gpu/imx/dcss/dcss-hdr10-tables.h
index 93bcf68..ee18208 100644
--- a/drivers/gpu/imx/dcss/dcss-hdr10-tables.h
+++ b/drivers/gpu/imx/dcss/dcss-hdr10-tables.h
@@ -12,6 +12,7 @@
  * for more details.
  */
 
+
 #ifndef __DCSS_HDR10_TABLES_H__
 #define __DCSS_HDR10_TABLES_H__
 
@@ -57,7 +58,7 @@
 
 static u32 dcss_hdr10_tables[] = {
 	/* table descriptor */
-	0xe1fffff9, 0xf,
+	0xe1ffffe9, 0xf,
 	/* table length */
 	0x400,
 	/* table data */
@@ -176,7 +177,7 @@
 	0x390d, 0x391b, 0x392a, 0x3939, 0x3949, 0x3959, 0x3969, 0x3979, 0x398a,
 	0x399b, 0x39ac, 0x39be, 0x39d0, 0x39e2, 0x39f5, 0xc,
 	/* table descriptor */
-	0xe2fffff9, 0xf,
+	0xe2ffffe9, 0xf,
 	/* table length */
 	0x400,
 	/* table data */
@@ -295,7 +296,7 @@
 	0x39c5, 0x39c9, 0x39cd, 0x39d1, 0x39d5, 0x39d8, 0x39dc, 0x39e0, 0x39e4,
 	0x39e8, 0x39ec, 0x39f0, 0x39f4, 0x39f8, 0x39fc, 0xc,
 	/* table descriptor */
-	0xe4fffff9, 0xf,
+	0xe4ffffe9, 0xf,
 	/* table length */
 	0x400,
 	/* table data */
@@ -414,7 +415,7 @@
 	0x39ba, 0x39bf, 0x39c4, 0x39c8, 0x39cd, 0x39d1, 0x39d6, 0x39db, 0x39e0,
 	0x39e4, 0x39e9, 0x39ee, 0x39f3, 0x39f7, 0x39fc, 0xc,
 	/* table descriptor */
-	0xe8fffff9, 0xf,
+	0xe8ffffe9, 0xf,
 	/* table length */
 	0x400,
 	/* table data */
@@ -533,7 +534,7 @@
 	0x396c, 0x3975, 0x397f, 0x3989, 0x3992, 0x399c, 0x39a6, 0x39b0, 0x39ba,
 	0x39c4, 0x39cf, 0x39d9, 0x39e4, 0x39ee, 0x39f9, 0xc,
 	/* table descriptor */
-	0xf0fffff9, 0xf,
+	0xf0ffffe9, 0xf,
 	/* table length */
 	0x400,
 	/* table data */
@@ -652,7 +653,7 @@
 	0x39be, 0x39c2, 0x39c7, 0x39cb, 0x39cf, 0x39d4, 0x39d8, 0x39dd, 0x39e1,
 	0x39e6, 0x39ea, 0x39ef, 0x39f3, 0x39f8, 0x39fc, 0xc,
 	/* table descriptor */
-	0xbf7ffffa, 0xf,
+	0xbf7fffea, 0xf,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -660,7 +661,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff, 0x3ff, 0xe, 0x40, 0x40, 0x40, 0x40,
 	0x40, 0x40, 0x3ac, 0x3ac, 0x3ac,
 	/* table descriptor */
-	0xdf7ffffa, 0xf,
+	0xdf7fffea, 0xf,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -668,7 +669,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff, 0x3ff, 0xe, 0x0, 0x0, 0x0, 0x0, 0x0,
 	0x0, 0x3ff, 0x3ff, 0x3ff,
 	/* table descriptor */
-	0xbfbffffa, 0x0,
+	0xbfbfffea, 0x0,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -676,7 +677,7 @@
 	0xffffe63a, 0xfffffdbf, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff,
 	0x3ff, 0xe, 0x40, 0x200, 0x200, 0x40, 0x40, 0x40, 0x3ac, 0x3c0, 0x3c0,
 	/* table descriptor */
-	0x3dbffffa, 0x1,
+	0x3fbfffea, 0x1,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -684,7 +685,7 @@
 	0xffffe68b, 0xfffffd6e, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff,
 	0x3ff, 0xe, 0x40, 0x200, 0x200, 0x40, 0x40, 0x40, 0x3ac, 0x3c0, 0x3c0,
 	/* table descriptor */
-	0x3fbffffa, 0x6,
+	0x3fbfffea, 0x6,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -692,7 +693,7 @@
 	0xffffe888, 0xfffffb71, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff,
 	0x3ff, 0xe, 0x40, 0x200, 0x200, 0x40, 0x40, 0x40, 0x3ac, 0x3c0, 0x3c0,
 	/* table descriptor */
-	0x3fbffffa, 0x8,
+	0x3fbfffea, 0x8,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -700,7 +701,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff, 0x3ff, 0xe, 0x40, 0x200, 0x200, 0x40,
 	0x40, 0x40, 0x3ac, 0x3c0, 0x3c0,
 	/* table descriptor */
-	0xdfbffffa, 0x0,
+	0xdfbfffea, 0x0,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -708,7 +709,7 @@
 	0xffffe293, 0xfffffd6d, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff,
 	0x3ff, 0xe, 0x0, 0x200, 0x200, 0x0, 0x0, 0x0, 0x3ff, 0x3ff, 0x3ff,
 	/* table descriptor */
-	0x5dbffffa, 0x1,
+	0x5fbfffea, 0x1,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -716,7 +717,7 @@
 	0xffffe2ef, 0xfffffd11, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff,
 	0x3ff, 0xe, 0x0, 0x200, 0x200, 0x0, 0x0, 0x0, 0x3ff, 0x3ff, 0x3ff,
 	/* table descriptor */
-	0x5fbffffa, 0x6,
+	0x5fbfffea, 0x6,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -724,7 +725,7 @@
 	0xffffe534, 0xfffffacc, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff,
 	0x3ff, 0xe, 0x0, 0x200, 0x200, 0x0, 0x0, 0x0, 0x3ff, 0x3ff, 0x3ff,
 	/* table descriptor */
-	0x5fbffffa, 0x8,
+	0x5fbfffea, 0x8,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -732,7 +733,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff, 0x3ff, 0xe, 0x0, 0x200, 0x200, 0x0,
 	0x0, 0x0, 0x3ff, 0x3ff, 0x3ff,
 	/* table descriptor */
-	0xfff0e1f1, 0xf,
+	0xfff0e1e1, 0xf,
 	/* table length */
 	0x402,
 	/* table data */
@@ -851,7 +852,7 @@
 	0x3979, 0x3981, 0x398a, 0x3992, 0x399b, 0x39a3, 0x39ac, 0x39b5, 0x39be,
 	0x39c7, 0x39d0, 0x39d9, 0x39e2, 0x39ec, 0x39f5, 0x39ff, 0x3,
 	/* table descriptor */
-	0xffffe2f1, 0xf,
+	0xffffe2e1, 0xf,
 	/* table length */
 	0x402,
 	/* table data */
@@ -964,7 +965,7 @@
 	0x3efd, 0x3f1d, 0x3f3d, 0x3f5d, 0x3f7d, 0x3f9e, 0x3fbe, 0x3fde, 0x3fff,
 	0x0,
 	/* table descriptor */
-	0xffffe4f1, 0xf,
+	0xffffe4e1, 0xf,
 	/* table length */
 	0x402,
 	/* table data */
@@ -1073,7 +1074,7 @@
 	0x3dc4, 0x3dea, 0x3e0f, 0x3e35, 0x3e5b, 0x3e81, 0x3ea7, 0x3ecd, 0x3ef3,
 	0x3f19, 0x3f3f, 0x3f65, 0x3f8b, 0x3fb2, 0x3fd8, 0x3fff, 0x0,
 	/* table descriptor */
-	0xfff0e8f1, 0xf,
+	0xfff0e8e1, 0xf,
 	/* table length */
 	0x402,
 	/* table data */
@@ -1192,7 +1193,7 @@
 	0x39ab, 0x39b0, 0x39b5, 0x39ba, 0x39bf, 0x39c4, 0x39c9, 0x39cf, 0x39d4,
 	0x39d9, 0x39de, 0x39e4, 0x39e9, 0x39ee, 0x39f4, 0x39f9, 0x39ff, 0x3,
 	/* table descriptor */
-	0xfffff0f1, 0xf,
+	0xfffff0e1, 0xf,
 	/* table length */
 	0x402,
 	/* table data */
@@ -1303,92 +1304,6 @@
 	0x3e4d, 0x3e71, 0x3e95, 0x3eb9, 0x3edd, 0x3f01, 0x3f25, 0x3f49, 0x3f6d,
 	0x3f91, 0x3fb6, 0x3fda, 0x3fff, 0x0,
 	/* table descriptor */
-	0xffffbf52, 0xf,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x4abd, 0x0, 0x0, 0x0, 0x4abd, 0x0, 0x0, 0x0, 0x4abd, 0xfffffff0,
-	0xfffffff0, 0xfffffff0, 0x0, 0x0, 0x0, 0xdb, 0xdb, 0xdb, 0xc, 0x0, 0x0,
-	0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff, 0x3ff,
-	/* table descriptor */
-	0xffffdf52, 0xf,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x4030, 0x0, 0x0, 0x0, 0x4030, 0x0, 0x0, 0x0, 0x4030, 0x0, 0x0,
-	0x0, 0x0, 0x0, 0x0, 0xff, 0xff, 0xff, 0xc, 0x0, 0x0, 0x0, 0x0, 0x0,
-	0x0, 0x3ff, 0x3ff, 0x3ff,
-	/* table descriptor */
-	0xfff0bf92, 0xf,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x255f, 0x0, 0x35e0, 0x255f, 0xfffff9fd, 0xffffeb20, 0x255f,
-	0x44bd, 0x0, 0xfffffff0, 0xffffff80, 0xffffff80, 0x0, 0xffffff90,
-	0xffffff90, 0xdb, 0x70, 0x70, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff,
-	0x3ff, 0x3ff,
-	/* table descriptor */
-	0xfff13692, 0xf,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x255f, 0x0, 0x3989, 0x255f, 0xfffff928, 0xffffeee6, 0x255f,
-	0x43cc, 0x0, 0xfffffff0, 0xffffff80, 0xffffff80, 0x0, 0xffffff90,
-	0xffffff90, 0xdb, 0x70, 0x70, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff,
-	0x3ff, 0x3ff,
-	/* table descriptor */
-	0xfff63692, 0xf,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x255f, 0x0, 0x3339, 0x255f, 0xfffff36d, 0xffffe5e9, 0x255f,
-	0x40be, 0x0, 0xfffffff0, 0xffffff80, 0xffffff80, 0x0, 0xffffff90,
-	0xffffff90, 0xdb, 0x70, 0x70, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff,
-	0x3ff, 0x3ff,
-	/* table descriptor */
-	0xfff83692, 0xf,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x4abd, 0x0, 0x0, 0x0, 0x4912, 0x0, 0x0, 0x0, 0x4912, 0xfffffff0,
-	0xffffff80, 0xffffff80, 0x0, 0xffffff90, 0xffffff90, 0xdb, 0x70, 0x70,
-	0xc, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff, 0x3ff,
-	/* table descriptor */
-	0xfff0df92, 0xf,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x4030, 0x0, 0x5ea7, 0x4030, 0xfffff570, 0xffffdb53, 0x4030,
-	0x78c4, 0x0, 0x0, 0xffffff80, 0xffffff80, 0x0, 0xffffff80, 0xffffff80,
-	0xff, 0x7f, 0x7f, 0xc, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff,
-	0x3ff,
-	/* table descriptor */
-	0xfff15692, 0xf,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x4030, 0x0, 0x6515, 0x4030, 0xfffff3fa, 0xffffe1f4, 0x4030,
-	0x771c, 0x0, 0x0, 0xffffff80, 0xffffff80, 0x0, 0xffffff80, 0xffffff80,
-	0xff, 0x7f, 0x7f, 0xc, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff,
-	0x3ff,
-	/* table descriptor */
-	0xfff65692, 0xf,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x4030, 0x0, 0x59fe, 0x4030, 0xffffe9e9, 0xffffd229, 0x4030,
-	0x71be, 0x0, 0x0, 0xffffff80, 0xffffff80, 0x0, 0xffffff80, 0xffffff80,
-	0xff, 0x7f, 0x7f, 0xc, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff,
-	0x3ff,
-	/* table descriptor */
-	0xfff85692, 0xf,
-	/* table length */
-	0x1d,
-	/* table data */
-	0x3, 0x4030, 0x0, 0x0, 0x0, 0x4030, 0x0, 0x0, 0x0, 0x4030, 0x0,
-	0xffffff80, 0xffffff80, 0x0, 0xffffff80, 0xffffff80, 0xff, 0x7f, 0x7f,
-	0xc, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff, 0x3ff,
-	/* table descriptor */
 	0xffffbf62, 0xf,
 	/* table length */
 	0x1d,
@@ -1475,7 +1390,7 @@
 	0xfffffe00, 0xfffffe00, 0x0, 0xfffffe00, 0xfffffe00, 0x3ff, 0x1ff,
 	0x1ff, 0xe, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3ff, 0x3ff, 0x3ff,
 	/* table descriptor */
-	0xfff176f4, 0x0,
+	0xfff17664, 0x0,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1483,7 +1398,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0xfff276f4, 0x0,
+	0xfff27664, 0x0,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1491,7 +1406,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0xfff476f4, 0x0,
+	0xfff47664, 0x0,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1499,7 +1414,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0xfff876f4, 0x0,
+	0xfff87664, 0x0,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1507,7 +1422,39 @@
 	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff276f4, 0x1,
+	0xfff176a4, 0x0,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x503c, 0x2a1c, 0x58a, 0x8d6, 0x7598, 0x174, 0x219, 0xb41, 0x7288,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x1, 0x0, 0x0,
+	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
+	/* table descriptor */
+	0xfff276a4, 0x0,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x53c5, 0x26a4, 0x579, 0x93a, 0x7539, 0x170, 0x230, 0xc84, 0x712e,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x1, 0x0, 0x0,
+	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
+	/* table descriptor */
+	0xfff476a4, 0x0,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x4e04, 0x2c5e, 0x57f, 0xaa6, 0x713c, 0x400, 0x208, 0xa67, 0x7372,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x1, 0x0, 0x0,
+	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
+	/* table descriptor */
+	0xfff876a4, 0x0,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x7057, 0x9d1, 0x5b9, 0xd6e, 0x70f3, 0x180, 0x416, 0x574, 0x7659,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x1, 0x0, 0x0,
+	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
+	/* table descriptor */
+	0x7ff27664, 0x1,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1515,7 +1462,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff476f4, 0x1,
+	0x7ff47664, 0x1,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1524,7 +1471,7 @@
 	0x3fff, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000,
 	0xffc0000,
 	/* table descriptor */
-	0x7ff876f4, 0x1,
+	0x7ff87664, 0x1,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1532,7 +1479,32 @@
 	0x5818, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff176f4, 0x2,
+	0x7ff276a4, 0x1,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x42c2, 0xfffffd2f, 0x0, 0x0, 0x3ff1, 0x0, 0x0, 0xc1, 0x3f30, 0x0,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x0, 0x0, 0x0, 0x0,
+	0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
+	/* table descriptor */
+	0x7ff476a4, 0x1,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x3d92, 0x330, 0xffffff2f, 0x12a, 0x3d56, 0x171, 0xffffffe5,
+	0xffffffb9, 0x4053, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff,
+	0x3fff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000,
+	0xffc0000,
+	/* table descriptor */
+	0x7ff876a4, 0x1,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x592d, 0xffffe6c4, 0x0, 0x98, 0x3f59, 0x0, 0x97, 0xfffffd48,
+	0x4212, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x0, 0x0,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
+	/* table descriptor */
+	0x7ff17664, 0x2,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1540,7 +1512,7 @@
 	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff476f4, 0x2,
+	0x7ff47664, 0x2,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1549,7 +1521,7 @@
 	0x3fff, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000,
 	0xffc0000,
 	/* table descriptor */
-	0x7ff876f4, 0x2,
+	0x7ff87664, 0x2,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1557,7 +1529,32 @@
 	0x5925, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff176f4, 0x4,
+	0x7ff176a4, 0x2,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x3d3e, 0x2b3, 0x0, 0x0, 0x3ff1, 0x0, 0x0, 0xffffff3d, 0x40b4,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x0, 0x0, 0x0,
+	0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
+	/* table descriptor */
+	0x7ff476a4, 0x2,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x3b06, 0x5a4, 0xffffff47, 0x12a, 0x3d56, 0x171, 0xffffffe1,
+	0xfffffefd, 0x4113, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff,
+	0x3fff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000,
+	0xffc0000,
+	/* table descriptor */
+	0x7ff876a4, 0x2,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x5570, 0xffffea81, 0x0, 0x98, 0x3f59, 0x0, 0x97, 0xfffffc7e,
+	0x42dc, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x0, 0x0,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
+	/* table descriptor */
+	0x7ff17664, 0x4,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1565,7 +1562,7 @@
 	0x60, 0x54bd, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff,
 	0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff276f4, 0x4,
+	0x7ff27664, 0x4,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1573,7 +1570,7 @@
 	0x15f, 0x53be, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff,
 	0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff876f4, 0x4,
+	0x7ff87664, 0x4,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1582,15 +1579,32 @@
 	0x3fff, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000,
 	0xffc0000,
 	/* table descriptor */
-	0xfff0fff4, 0xf,
+	0x7ff176a4, 0x4,
 	/* table length */
 	0x1d,
 	/* table data */
-	0x8000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-	0x0, 0x0,
+	0x3, 0x4278, 0xfffffc8d, 0xec, 0xfffffebc, 0x42b8, 0xfffffe7d, 0x1b,
+	0x48, 0x3f8e, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff176f4, 0x8,
+	0x7ff276a4, 0x4,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x4566, 0xfffff9a2, 0xe9, 0xfffffeae, 0x42c1, 0xfffffe82, 0x1c,
+	0x107, 0x3ece, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
+	/* table descriptor */
+	0x7ff876a4, 0x4,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x5cae, 0xffffe250, 0xf4, 0xfffffed8, 0x42a9, 0xfffffe70, 0xbc,
+	0xfffffd89, 0x41ac, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff,
+	0x3fff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000,
+	0xffc0000,
+	/* table descriptor */
+	0x7ff17664, 0x8,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1598,7 +1612,7 @@
 	0x5282, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff276f4, 0x8,
+	0x7ff27664, 0x8,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1606,7 +1620,7 @@
 	0x5189, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7ff476f4, 0x8,
+	0x7ff47664, 0x8,
 	/* table length */
 	0x1d,
 	/* table data */
@@ -1614,13 +1628,53 @@
 	0x2cb, 0x5316, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff,
 	0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 	/* table descriptor */
-	0x7fff76f4, 0xf,
+	0x7fff7664, 0xf,
 	/* table length */
 	0x1d,
 	/* table data */
 	0x3, 0x5541, 0x0, 0x0, 0x0, 0x5541, 0x0, 0x0, 0x0, 0x5541, 0x0, 0x0,
 	0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x2, 0x0, 0x0, 0x0, 0x0,
-	0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000
+	0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
+	/* table descriptor */
+	0xfff0ffe4, 0xf,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x8000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
+	0x0, 0x0,
+	/* table descriptor */
+	0x7ff176a4, 0x8,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x2dba, 0x1237, 0x0, 0xffffff92, 0x405f, 0x0, 0xffffff93, 0x27d,
+	0x3de1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x0, 0x0,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
+	/* table descriptor */
+	0x7ff276a4, 0x8,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x2fbd, 0x1034, 0x0, 0xffffff8d, 0x4064, 0x0, 0xffffff8e, 0x33c,
+	0x3d27, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x0, 0x0,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
+	/* table descriptor */
+	0x7ff476a4, 0x8,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x58ba, 0x2781, 0xffffffa7, 0x185, 0x7b74, 0x2ea, 0xffffff11,
+	0x431, 0x7ca0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff,
+	0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
+	/* table descriptor */
+	0x7fff76a4, 0xf,
+	/* table length */
+	0x1d,
+	/* table data */
+	0x3, 0x7fe2, 0x0, 0x0, 0x0, 0x7fe2, 0x0, 0x0, 0x0, 0x7fe2, 0x0, 0x0,
+	0x0, 0x0, 0x0, 0x0, 0x3fff, 0x3fff, 0x3fff, 0x1, 0x0, 0x0, 0x0, 0x0,
+	0x0, 0x0, 0xffc0000, 0xffc0000, 0xffc0000,
 };
 
 #endif /* __DCSS_HDR10_TABLES_H__ */
-- 
1.7.9.5

